scipy/optimize/_qap.py

```diff
@@ -0,0 +1,776 @@
+        :ref:`'2opt' <optimize.qap-2opt>` is also available.
+
+    Options
+    -------
+
+    maximize : bool (default = False)
+        Setting `maximize` to ``True`` solves the Graph Matching Problem (GMP)
+        rather than the Quadratic Assingnment Problem (QAP). This is
+        accomplished through trivial negation of the objective function.
+
+    rng : {None, int, `~np.random.RandomState`, `~np.random.Generator`}
+        This parameter defines the object to use for drawing random
+        variates.
+        If `rng` is ``None`` the `~np.random.RandomState` singleton is
+        used.
+        If `rng` is an int, a new ``RandomState`` instance is used,
+        seeded with `rng`.
+        If `rng` is already a ``RandomState`` or ``Generator``
+        instance, then that object is used.
+        Default is None.
+
+    partial_match : 2d-array of integers, optional, (default = None)
+        Allows the user to fix part of the matching between the two
+        matrices. In the literature, a partial match is also known as a
+        "seed".
+
+        Each row of `partial_match` specifies the indices of a pair of
+        corresponding nodes, that is, node ``partial_match[i, 0]`` of `A` is
+        matched to node ``partial_match[i, 1]`` of `B`. Accordingly,
+        ``partial_match`` is an array of size ``(m , 2)``, where ``m`` is
+        not greater than the number of nodes.
+
+    P0 : 2d-array, "barycenter", or "randomized" (default = "barycenter")
+        The initial (guess) permutation matrix or search "position"
+        :math:`P0`.
+
+        :math:`P0` need not be a proper permutation matrix;
+        however, it must have the same shape as `A` and
+        `B`, and it must be doubly stochastic: each of its
+        rows and columns must sum to 1.
+
+        If unspecified or ``"barycenter"``, the non-informative "flat
+        doubly stochastic matrix" :math:`J = 1*1^T/n`, where :math:`n`
+        is the number of nodes and :math:`1` is a :math:`n \times 1`
+        array of ones, is used. This is the "barycenter" of the
+        search space of doubly-stochastic matrices.
+
+        If ``"randomized"``, the algorithm will start from the
+        randomized initial search position :math:`P_0 = (J + K)/2`,
+        where :math:`J` is the "barycenter" and :math:`K` is a random
+        doubly stochastic matrix.
+
+    shuffle_input : bool (default = False)
+        To avoid artificially high or low matching due to inherent
+        sorting of input matrices, gives users the option
+        to shuffle the nodes. Results are then unshuffled so that the
+        returned results correspond with the node order of inputs.
+        Shuffling may cause the algorithm to be non-deterministic,
+        unless a random seed is set or an `rng` option is provided.
+
+    maxiter : int, positive (default = 30)
+        Integer specifying the max number of Franke-Wolfe iterations performed.
+
+    tol : float (default = 0.05)
```

Comment:
Looking at this again I see that this tolerance is absolute. The standard in `scipy.optimize` appears to be `atol` for absolute tolerances and `tol` for relative tolerances.

We could either rename the parameter to `atol`, or define a relative tolerance as `(P_{i} - P_{i+1}) / sqrt(len(P))`. The denominator `sqrt(P)` appears to work better than `np.linalg.norm(P)` because the norm of a doubly-stochastic matrix can vary quite significantly, but the norm of a permutation matrix is always `sqrt(len(P))`.

Advantage of renaming to `atol`: simplicity.
Advantage of defining a relative tolerance: invariance to the input matrix size.

Reply:
There is a standard in `np.testing` to distinguish `rtol` from`atol`, but it is hardly consistent throughput `scipy.optimize`. `tol` is the most common name for the option, but each method defines what it does. We can do anything here, including leave it alone, but making it a relative tolerance would be an objective improvement.

Reply:
I can make this change and PR it to your branch @mdhaber